From da164b7bcbde9b2cfb7c7909dfe6956977c321b8 Mon Sep 17 00:00:00 2001
From: Gu Chao <gux.chao@intel.com>
Date: Wed, 24 Jun 2020 16:30:01 +0800
Subject: [PATCH] DRV_DPM_SW-292: FLM DPM remove function param subif_name from
 dp_inst_add_dev() and dp_inst_del_dev()

- remove subif_name from struct dp_dev
---
 drivers/net/datapath/dpm/datapath_api.c         | 10 +++----
 drivers/net/datapath/dpm/datapath_instance.c    | 40 ++++++++-----------------
 drivers/net/datapath/dpm/datapath_instance.h    |  5 ++--
 drivers/net/datapath/dpm/datapath_logical_dev.c |  6 ++--
 4 files changed, 21 insertions(+), 40 deletions(-)

diff --git a/drivers/net/datapath/dpm/datapath_api.c b/drivers/net/datapath/dpm/datapath_api.c
index fffef71b2d37..f7b1b0fc1852 100644
--- a/drivers/net/datapath/dpm/datapath_api.c
+++ b/drivers/net/datapath/dpm/datapath_api.c
@@ -786,7 +786,7 @@ int32_t dp_register_subif_private(int inst, struct module *owner,
 	port_info->num_subif++;
 	if (get_dp_bp_info(inst, sif->bp)->ref_cnt > 1)
 		return DP_SUCCESS;
-	dp_inst_add_dev(dev, subif_name, subif_id->inst, subif_id->port_id,
+	dp_inst_add_dev(dev, subif_id->inst, subif_id->port_id,
 			sif->bp, subif_id->subif, flags);
 	return DP_SUCCESS;
 }
@@ -879,8 +879,7 @@ int32_t dp_deregister_subif_private(int inst, struct module *owner,
 		DP_DEBUG(DP_DBG_FLAG_REG, "%s for %s inst=%d bp=%d\n",
 			 "dp_inst_del_dev", dev->name, inst, bp);
 
-		dp_inst_del_dev(dev, subif_name, inst, port_id,
-				subif_id->subif, 0);
+		dp_inst_del_dev(dev, inst, port_id, subif_id->subif, 0);
 
 		dp_notifier_invoke(inst, dev, port_id,
 				   DP_EVENT_DE_REGISTER_SUBIF);
@@ -1064,7 +1063,7 @@ int32_t dp_update_subif_info(struct dp_subif_upd_info *info)
 				bp_info->ref_cnt = 1;
 				bp_info->flag = 1;
 			}
-			dp_inst_add_dev(info->new_dev, NULL, info->inst,
+			dp_inst_add_dev(info->new_dev, info->inst,
 					info->dp_port, bp, vap, 0);
 		} else {
 			DP_DEBUG(DP_DBG_FLAG_REG,
@@ -1116,8 +1115,7 @@ int32_t dp_update_subif_info(struct dp_subif_upd_info *info)
 		DP_DEBUG(DP_DBG_FLAG_REG,
 			 "del old master dev(%s) from dp_dev list\n",
 			 old_dev->name);
-		dp_inst_del_dev(old_dev, NULL, info->inst, info->dp_port,
-				vap, 0);
+		dp_inst_del_dev(old_dev, info->inst, info->dp_port, vap, 0);
 
 		dp_notifier_invoke(info->inst, old_dev, info->dp_port,
 				   DP_EVENT_DE_REGISTER_SUBIF);
diff --git a/drivers/net/datapath/dpm/datapath_instance.c b/drivers/net/datapath/dpm/datapath_instance.c
index a99a3ad779b4..f822110139ff 100644
--- a/drivers/net/datapath/dpm/datapath_instance.c
+++ b/drivers/net/datapath/dpm/datapath_instance.c
@@ -223,13 +223,7 @@ static int dp_ndo_setup_tc(struct net_device *dev,
 #endif /* LINUX_VERSION_CODE */
 #endif /* CONFIG_QOS_MGR */
 
-/*Note:
- *dev and subif_name: only one will be used for the hash index calculation.
- *subif_name is only used for ATM IPOA/PPPOA case since its dev is NULL.
- *otherwise it should use its dev.
- *it will not work if just dev->name as subif_name
- */
-int dp_inst_add_dev(struct net_device *dev, char *subif_name, int inst,
+int dp_inst_add_dev(struct net_device *dev, int inst,
 		    int ep, int bp, int ctp, u32 flag)
 {
 	struct dp_dev *dp_dev;
@@ -260,12 +254,6 @@ int dp_inst_add_dev(struct net_device *dev, char *subif_name, int inst,
 		dp_dev->ctp = ctp;
 		dp_dev->ep = ep;
 		dp_dev->bp = bp;
-
-		if (subif_name) {
-			strncpy(dp_dev->subif_name, subif_name,
-				sizeof(dp_dev->subif_name) - 1);
-		}
-
 		if (new_f) {
 			idx = dp_dev_hash(dev);
 			hlist_add_head(&dp_dev->hlist, &dp_dev_list[idx]);
@@ -294,7 +282,7 @@ int dp_inst_add_dev(struct net_device *dev, char *subif_name, int inst,
 	return DP_SUCCESS;
 }
 
-int dp_inst_del_dev(struct net_device *dev, char *subif_name, int inst,
+int dp_inst_del_dev(struct net_device *dev, int inst,
 		    int ep, u16 ctp, u32 flag)
 {
 	struct dp_dev *dp_dev;
@@ -307,23 +295,23 @@ int dp_inst_del_dev(struct net_device *dev, char *subif_name, int inst,
 
 	dp_dev = dp_dev_lookup(dev);
 	if (!dp_dev) {
-		pr_err("Failed to dp_dev_lookup: %s_%s\n",
-		       dev ? dev->name : "NULL", subif_name);
+		pr_err("Failed to dp_dev_lookup: %s\n",
+		       dev ? dev->name : "NULL");
 		return -1;
 	}
 	if (dp_dev->count <= 0) {
-		pr_err("Count(%d) should > 0(%s_%s)\n", dp_dev->count,
-		       dev ? dev->name : "NULL", subif_name);
+		pr_err("Count(%d) should > 0(%s)\n", dp_dev->count,
+		       dev ? dev->name : "NULL");
 		return -1;
 	}
 	if (dp_dev->inst != inst) {
-		pr_err("Why inst not same:%d_%d(%s_%s)\n", dp_dev->inst, inst,
-		       dev ? dev->name : "NULL", subif_name);
+		pr_err("Why inst not same:%d_%d(%s)\n", dp_dev->inst, inst,
+		       dev ? dev->name : "NULL");
 		return -1;
 	}
 	if (dp_dev->ep != ep) {
-		pr_err("Why ep not same:%d_%d(%s_%s)\n", dp_dev->ep, ep,
-		       dev ? dev->name : "NULL", subif_name);
+		pr_err("Why ep not same:%d_%d(%s)\n", dp_dev->ep, ep,
+		       dev ? dev->name : "NULL");
 		return -1;
 	}
 		/* Check added to remove pmapper device from dp_dev list
@@ -374,9 +362,8 @@ int dp_inst_del_dev(struct net_device *dev, char *subif_name, int inst,
 		return 0;
 	}
 
-	DP_DEBUG(DP_DBG_FLAG_INST, "dp_del_dev fail(%s_%s): not found ctp=%d\n",
-		 dev ? dev->name : "NULL",
-		 subif_name, ctp);
+	DP_DEBUG(DP_DBG_FLAG_INST, "dp_del_dev fail(%s): not found ctp=%d\n",
+		 dev ? dev->name : "NULL", ctp);
 	return -1;
 }
 
@@ -629,8 +616,7 @@ int proc_inst_dev_dump(struct seq_file *s, int pos)
 		   "pos",
 		   pos,
 		   "dev",
-		   dp_dev_proc->dev ? dp_dev_proc->dev->name :
-		   dp_dev_proc->subif_name,
+		   dp_dev_proc->dev ? dp_dev_proc->dev->name : "NULL",
 		   dp_dev_proc->dev,
 		   "inst",
 		   dp_dev_proc->inst,
diff --git a/drivers/net/datapath/dpm/datapath_instance.h b/drivers/net/datapath/dpm/datapath_instance.h
index 24bc94cd5f8f..4101a32da15a 100644
--- a/drivers/net/datapath/dpm/datapath_instance.h
+++ b/drivers/net/datapath/dpm/datapath_instance.h
@@ -37,7 +37,6 @@ struct dp_mod {
 struct dp_dev {
 	struct hlist_node hlist;
 	struct net_device *dev;
-	char subif_name[IFNAMSIZ]; /*for ATM IPOA/PPPOA */
 	int inst;
 	int ep;
 	int bp;
@@ -61,9 +60,9 @@ int dp_get_inst_via_dev(struct net_device *dev,
 			char *subif_name, u32 flag);
 int dp_get_inst_via_module(struct module *owner, u16 ep, u32 flag);
 struct dp_hw_cap *match_hw_cap(struct dp_inst_info *info, u32 flag);
-int dp_inst_add_dev(struct net_device *dev, char *subif_name, int inst,
+int dp_inst_add_dev(struct net_device *dev, int inst,
 		    int ep, int bp, int ctp, u32 flag);
-int dp_inst_del_dev(struct net_device *dev, char *subif_name, int inst,
+int dp_inst_del_dev(struct net_device *dev, int inst,
 		    int ep, u16 ctp, u32 flag);
 int dp_inst_insert_mod(struct module *owner, u16 ep, u32 inst, u32 flag);
 int dp_inst_del_mod(struct module *owner, u16 ep, u32 flag);
diff --git a/drivers/net/datapath/dpm/datapath_logical_dev.c b/drivers/net/datapath/dpm/datapath_logical_dev.c
index 56d9615480c3..b883329b638a 100644
--- a/drivers/net/datapath/dpm/datapath_logical_dev.c
+++ b/drivers/net/datapath/dpm/datapath_logical_dev.c
@@ -158,9 +158,7 @@ int add_logic_dev(int inst, int port_id, struct net_device *dev,
 
 	subif_id->bport = logic_dev_tmp->bp;
 	subif_id->subif = subif.subif;
-	dp_inst_add_dev(dev, NULL,
-			inst, port_id,
-			logic_dev_tmp->bp,
+	dp_inst_add_dev(dev, inst, port_id, logic_dev_tmp->bp,
 			subif.subif, flags);
 	return 0;
 }
@@ -179,7 +177,7 @@ int del_logic_dev(int inst, struct list_head *head, struct net_device *dev,
 	}
 	dp_info->subif_platform_set_unexplicit(inst, logic_dev->ep, logic_dev,
 					       flags);
-	dp_inst_del_dev(dev, NULL, inst, logic_dev->ep, logic_dev->ctp, flags);
+	dp_inst_del_dev(dev, inst, logic_dev->ep, logic_dev->ctp, flags);
 	list_del(&logic_dev->list);
 	devm_kfree(&g_dp_dev->dev, logic_dev);
 
